File: crm/test_logs.js
const fs = require('fs');
const txt = fs.readFileSync('src/lib/supabase.ts', 'utf8');
const url = txt.match(/const supabaseUrl = ['\"](https:\/\/[^'\"]+)['\"]/)[1];
const key = txt.match(/const supabaseAnonKey = ['\"](eyJ[^'\"]+)['\"]/)[1];
const headers = { apikey: key, Authorization: 'Bearer ' + key };

async function run() {
  const r1 = await fetch(url + '/rest/v1/leads?select=id,name,phone,source&order=created_at.desc&limit=5', {headers});
  console.log('Leads:', JSON.stringify(await r1.json(), null, 2));
  
  const r2 = await fetch(url + '/rest/v1/conversations?select=role,content,lead_id&order=created_at.desc&limit=4', {headers});
  console.log('Conv:', JSON.stringify(await r2.json(), null, 2));
}
run();

File: crm/test_logs.py
import urllib.request, json, re

with open('src/lib/supabase.ts', 'r', encoding='utf-8') as f:
    txt = f.read()

url = re.search(r"const supabaseUrl = ['\"](https://[^'\"]+)['\"]", txt).group(1)
key = re.search(r"const supabaseAnonKey = ['\"](eyJ[^'\"]+)['\"]", txt).group(1)

req = urllib.request.Request(f"{url}/rest/v1/conversations?select=role,content,lead_id&order=created_at.desc&limit=5", headers={
    'apikey': key,
    'Authorization': f'Bearer {key}'
})

with urllib.request.urlopen(req) as response:
    data = json.loads(response.read().decode())
    print("--- CONVERSATIONS ---")
    for row in data:
       print(f"Role: {row['role']}, Lead: {row['lead_id']}")
       print(f"[{row['content']}]")
       print("-----")
